blockchain/fullblocktests/error.go

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2022 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package fullblocktests
+
+// ErrorKind identifies a kind of error.  It has full support for errors.Is and
+// errors.As, so the caller can directly check against an error kind when
+// determining the reason for an error.
+type ErrorKind string
+
+// These constants are used to identify a specific ErrorKind.
+const (
+	// ErrDuplicateBlock indicates a block with the same hash already exists.
+	ErrDuplicateBlock = ErrorKind("ErrDuplicateBlock")
+
+	// ErrBlockTooBig indicates the serialized block size exceeds the maximum
+	// allowed size.
+	ErrBlockTooBig = ErrorKind("ErrBlockTooBig")
+
+	// ErrWrongBlockSize indicates that the block size in the header is not the
+	// actual serialized size of the block.
+	ErrWrongBlockSize = ErrorKind("ErrWrongBlockSize")
+
+	// ErrInvalidTime indicates the time in the passed block has a precision
+	// that is more than one second.  The chain consensus rules require
+	// timestamps to have a maximum precision of one second.
+	ErrInvalidTime = ErrorKind("ErrInvalidTime")
+
+	// ErrTimeTooOld indicates the time is either before the median time of the
+	// last several blocks per the chain consensus rules or prior to the most
+	// recent checkpoint.
+	ErrTimeTooOld = ErrorKind("ErrTimeTooOld")
+
+	// ErrTimeTooNew indicates the time is too far in the future as compared the
+	// current time.
+	ErrTimeTooNew = ErrorKind("ErrTimeTooNew")
+
+	// ErrUnexpectedDifficulty indicates specified bits do not align with the
+	// expected value either because it doesn't match the calculated value based
+	// on difficulty regarding the rules or it is out of the valid range.
+	ErrUnexpectedDifficulty = ErrorKind("ErrUnexpectedDifficulty")
+
+	// ErrHighHash indicates the block does not hash to a value which is lower
+	// than the required target difficultly.
+	ErrHighHash = ErrorKind("ErrHighHash")
+
+	// ErrBadMerkleRoot indicates the calculated merkle root does not match the
+	// expected value.
+	ErrBadMerkleRoot = ErrorKind("ErrBadMerkleRoot")
+
+	// ErrNoTransactions indicates the block does not have a least one
```

Comment:
a least -> at least

Reply:
I had just copied these over from the existing ones.  Good catch on the typos.  I've updated the old ones as well as new ones for all of these.